New Bank/NewBank/newbank/server/NewBank.java

```diff
@@ -82,4 +107,17 @@ private String moveFunds(CustomerID customer,double amount,String from,String to
 		}
 		return "FAIL";
 	}
+
+	private String payment(Customer customer, double amount,String from,String to,  Customer custCredit, String debitBankName, String creditBankName ) {
+		if(amount > 0) { // should never need to move a negative amount
```

Comment:
It might be worth having an additional check here to see if the customer exits in the Customers array. For example, if you misspell John as Jonh it should return false. Otherwise it gets stuck in a null loop.

Reply:
added